homeassistant/components/esphome/assist_satellite.py

```diff
@@ -127,27 +127,47 @@ def __init__(self, entry: ESPHomeConfigEntry) -> None:
             available_wake_words=[], active_wake_words=[], max_active_wake_words=1
         )
 
-    @property
-    def pipeline_entity_id(self) -> str | None:
-        """Return the entity ID of the pipeline to use for the next conversation."""
-        assert self._entry_data.device_info is not None
+        self._secondary_pipeline_active = False
+
+    def _get_entity_id(self, suffix: str) -> str | None:
+        """Return the entity id for pipeline select, etc."""
+        if self._entry_data.device_info is None:
+            return None
+
         ent_reg = er.async_get(self.hass)
         return ent_reg.async_get_entity_id(
             Platform.SELECT,
             DOMAIN,
-            f"{self._entry_data.device_info.mac_address}-pipeline",
+            f"{self._entry_data.device_info.mac_address}-{suffix}",
         )
 
+    @property
+    def pipeline_entity_id(self) -> str | None:
+        """Return the entity ID of the primary pipeline to use for the next conversation."""
+        if self._secondary_pipeline_active:
```

Comment:
I know we're only supporting 2 right now, but can we already make this work with a function that receives an index ?

Reply:
`get_pipeline_entity(self, index: int) -> str | None`